include/grpc/grpc_security.h

```diff
@@ -324,11 +332,16 @@ GRPCAPI gpr_timespec grpc_max_auth_token_lifetime(void);
    - json_key is the JSON key string containing the client's private key.
    - token_lifetime is the lifetime of each Json Web Token (JWT) created with
      this credentials.  It should not exceed grpc_max_auth_token_lifetime or
-     will be cropped to this value.  */
+     will be cropped to this value.
+   - user_provided_scope is an optional field for user to use in the JWT token
+     to represent the scope field.
+   - should_clear_audience dictating the clearance of audience field when it
+     is set to a non-zero value AND user_provided_scope is not nullptr.
+ */
 GRPCAPI grpc_call_credentials*
-grpc_service_account_jwt_access_credentials_create(const char* json_key,
-                                                   gpr_timespec token_lifetime,
-                                                   void* reserved);
+grpc_service_account_jwt_access_credentials_create(
+    const char* json_key, gpr_timespec token_lifetime,
+    const char* user_provided_scope, int should_clear_audience);
```

Comment:
We should rename `should_clear_audience` to `enable_scope_for_jwt` which is defined in the [aip](google.aip.dev/auth/4111).

Reply:
Note that this API is not meant to be used for only accessing the google API. Only the jwt token created inside `google_default_credentials_create()` will be used to access the google API. So `enable_scope_for_jwt` does not capture what `should_clear_audience` currently implies (i.e., clearing the audience field if it is set to true).

Reply:
So in this PR, we plan to support the following use cases: 
-  allow to specify both `audience` and `scope` fields in a JWT token, and it is controlled by `should_clear_audience = false`. 
- allow to include only `scope` field in a JWT token dictated by https://google.aip.dev/auth/4111, which will solve the regional endpoint issue, and it is controlled by `should_clear_audience = true`. 

It is not clear which case it refers to if using `enable_scope_for_jwt`. WDYT?

Reply:
I see, that makes sense.